lib/mongoid/document.rb

```diff
@@ -281,20 +296,39 @@ module ClassMethods
       # @param [ Hash ] attrs The hash of attributes to instantiate with.
       # @param [ Integer ] selected_fields The selected fields from the
       #   criteria.
+      # @param [ true | false ] defer_callbacks Flag specifies whether callbacks
+      #   should be run.
       #
       # @return [ Document ] A new document.
-      def instantiate(attrs = nil, selected_fields = nil)
+      def instantiate(attrs = nil, selected_fields = nil, defer_callbacks = false)
         attributes = attrs || {}
         doc = allocate
         doc.__selected_fields = selected_fields
         doc.instance_variable_set(:@attributes, attributes)
-        doc.apply_defaults
-        yield(doc) if block_given?
-        doc.run_callbacks(:find) unless doc._find_callbacks.empty?
-        doc.run_callbacks(:initialize) unless doc._initialize_callbacks.empty?
+
+        if defer_callbacks
+          doc.pending_callbacks.push(*[:default, :find, :initialize])
```

Comment:
What's the reason to splat the array here?

Can't this just be:

```
doc.pending_callbacks.push(:default, :find, :initialize)
```

Reply:
Yup, good call.